### PR TITLE
Clarify deprecation status of the async driver.

### PR DESCRIPTION
**Pull request by jyemin**
JAVARS-142

### CODE REVIEW

File: docs/reference/content/index.md [hunk 1]
```diff
@@ -8,10 +8,12 @@ type = "index"
...
 
-The Reactive Streams implementation is built upon the 
-[`MongoDB Async Driver`]({{< asyncDriver >}}) and we mirror their API. 
-For reference documentation please see the async driver documentation.
+The Reactive Streams implementation is built upon the callback-based [`MongoDB Async Driver`]({{< asyncDriver >}}) and mirrors its API. 
+For reference documentation please see the asynchronous driver documentation.
 
+Note that while the callback-based asynchronous driver is now deprecated, the MongoDB Reactive Streams Java 
+driver will continue to be supported. Once the callback-based driver has been removed, the MongoDB Reactive Streams Java 
+driver will be fully documented in its place
```
Comment (slhommed) [inline docs/reference/content/index.md]: I don't understand why we can't fully document the Reactive driver now.

Comment (jyemin) [reply docs/reference/content/index.md]: We could, but it will take a lot more time than this PR.

Comment (jyemin) [reply docs/reference/content/index.md]: Also, those docs will be moved soon to the Java driver github pages, so any links people create to docs we add now will be stale in a few months.

**Comment by jyemin** [comment]
@slhommed I improved the prose according to your suggestion.

**Pull request closed without merging**